safe_transaction_service/history/management/commands/setup_service.py

```diff
@@ -5,37 +5,67 @@
 from django.core.management.base import BaseCommand
 from django.db.models import Min
 
-from django_celery_beat.models import IntervalSchedule, PeriodicTask
+from django_celery_beat.models import CrontabSchedule, IntervalSchedule, PeriodicTask
 
 from gnosis.eth import EthereumClientProvider
 from gnosis.safe.addresses import MASTER_COPIES, PROXY_FACTORIES
 
 from ...models import IndexingStatus, ProxyFactory, SafeMasterCopy
 
 
+@dataclass
+class CronDefinition:
+    minute: str = "*"
+    hour: str = "*"
+    day_of_week: str = "*"
+    day_of_month: str = "*"
+    month_of_year: str = "*"
+
+
 @dataclass
 class CeleryTaskConfiguration:
     name: str
     description: str
-    interval: int
-    period: str
+    interval: int = 0
+    period: str = None
+    cron: CronDefinition = None
     enabled: bool = True
 
     def create_task(self) -> Tuple[PeriodicTask, bool]:
-        interval_schedule, _ = IntervalSchedule.objects.get_or_create(
-            every=self.interval, period=self.period
-        )
-        periodic_task, created = PeriodicTask.objects.get_or_create(
-            task=self.name,
-            defaults={
+        if self.cron is None:
+            interval_schedule, _ = IntervalSchedule.objects.get_or_create(
+                every=self.interval, period=self.period
+            )
+            defaults = {
                 "name": self.description,
                 "interval": interval_schedule,
                 "enabled": self.enabled,
-            },
+            }
+        else:
+            crontab_schedule, _ = CrontabSchedule.objects.get_or_create(
+                minute=self.cron.minute,
+                hour=self.cron.hour,
+                day_of_week=self.cron.day_of_week,
+                day_of_month=self.cron.day_of_month,
+                month_of_year=self.cron.month_of_year,
+            )
+            defaults = {
+                "name": self.description,
+                "crontab": crontab_schedule,
+                "enabled": self.enabled,
+            }
+
+        periodic_task, created = PeriodicTask.objects.get_or_create(
+            task=self.name,
+            defaults=defaults,
         )
         if not created:
             periodic_task.name = self.description
-            periodic_task.interval = interval_schedule
+            if self.cron is None:
```

Comment:
Same that the previous comment, if you are checking for an interval check if the interval exists, don't asume there's an interval just because there's no cron